cmd/notary-signer/passwordwrap/interface.go

```diff
@@ -0,0 +1,19 @@
+package passwordwrap
+
+// Storage is the interface to store/retrieve password from any specific implementation type.
+type Storage interface {
+	// Get password from storage.
+	GetPassword(alias string) (string, error)
+
+	// Set password in storage.
+	SetPassword(alias string, newPassword string) error
+}
+
+// Protector is the interface to wrap/unwrap password using any specific implementation type.
+type Protector interface {
```

Comment:
@endophage 
Here is what I infer from the example above:
 - There will only be one interface, the `Storage` interface.
 - Protector will implement `Encrypt/Decrypt` apart from implementing the `Storage` interface.
 - So, in future all plugins will only implement the `Storage` interface.

Question:
 - My question is WRT, the implementation of `func (p Protector) decrypt(pass string) (string, error)`. In the default implementation, there is no need to implement `Decrypt` if the passphrase continues to be stored in clear in an environment variable. In case of a plugin implementing the `Storage` interface in future, it may or may not be a secure storage on its own. However, that decision has to be made in the plugin. So, in such case, the `Encrypt/Decrypt` in the `Protector` implementation will not be needed. (Which would not make it a Protector anymore). Please let me know if I am missing something in my understanding here.